File: scripts/harness_tick.py
#!/usr/bin/env python3
import argparse
import json
from pathlib import Path

from equal_rank_nudge_bot import evaluate_state_file


WAITING_STATUSES = {"planned", "proposal_pending", "waiting_reply"}


def main() -> None:
    p = argparse.ArgumentParser(description="하네스 상태 파일 틱 처리기")
    p.add_argument("--state-file", required=True)
    p.add_argument("--minutes-since-reply", type=int, required=True)
    p.add_argument("--seed", type=int)
    p.add_argument("--apply", action="store_true")
    args = p.parse_args()

    state_file = Path(args.state_file)
    payload = json.loads(state_file.read_text(encoding="utf-8"))

    nudge = payload.get("nudge", {})
    status = payload.get("status", "planned")

    if "proposal_pending" not in nudge:
        nudge["proposal_pending"] = status in WAITING_STATUSES
        payload["nudge"] = nudge
        if args.apply:
            state_file.write_text(json.dumps(payload, ensure_ascii=False, indent=2) + "\n", encoding="utf-8")

    result = evaluate_state_file(
        state_file=state_file,
        minutes_since_reply=args.minutes_since_reply,
        seed=args.seed,
        apply=args.apply,
    )
    print(json.dumps(result, ensure_ascii=False))


if __name__ == "__main__":
    main()

File: scripts/harness_tick.ts
import { readFileSync, writeFileSync } from "node:fs";
import { evaluateStateFile } from "./equal_rank_nudge_bot";

type Args = {
  stateFile: string;
  minutesSinceReply: number;
  seed?: number;
  apply: boolean;
};

function parseArgs(argv: string[]): Args {
  const toFind = (key: string): string | undefined => {
    const idx = argv.indexOf(`--${key}`);
    if (idx === -1 || idx + 1 >= argv.length) return undefined;
    return argv[idx + 1];
  };
  const has = (key: string): boolean => argv.includes(`--${key}`);

  const stateFile = toFind("state-file");
  if (!stateFile) {
    throw new Error("required argument missing: --state-file");
  }

  const minutes = Number(toFind("minutes-since-reply"));
  if (!Number.isFinite(minutes)) {
    throw new Error("required argument missing or invalid: --minutes-since-reply");
  }

  return {
    stateFile,
    minutesSinceReply: minutes,
    seed: toFind("seed") ? Number(toFind("seed")) : undefined,
    apply: has("apply"),
  };
}

function main(): void {
  const args = parseArgs(process.argv.slice(2));
  const payload = JSON.parse(readFileSync(args.stateFile, "utf8")) as Record<string, unknown>;
  const nudge = (payload.nudge as Record<string, unknown>) ?? {};
  const status = payload.status;

  if (!Object.prototype.hasOwnProperty.call(nudge, "proposal_pending")) {
    const shouldNudge = ["planned", "proposal_pending", "waiting_reply"].includes(String(status));
    nudge.proposal_pending = shouldNudge;
    const nextPayload = { ...payload, nudge };
    if (args.apply) {
      writeFileSync(args.stateFile, `${JSON.stringify(nextPayload, null, 2)}\n`, "utf8");
    }
  }

  const result = evaluateStateFile(args.stateFile, args.minutesSinceReply, args.seed, args.apply);
  process.stdout.write(`${JSON.stringify(result, null, 2)}\n`);
}

if (import.meta.main) {
  main();
}
